addons/project/models/project.py

```diff
@@ -2592,6 +2592,17 @@ def action_continue_recurrence(self):
         self.recurrence_id = False
         self.recurring_task = False
 
+    def action_menu_view_all_task(self):
+        user = self.env.user
+        stage = self.env['project.task.type'].sudo().search([('user_id', '=', user.id)], limit=1)
+        # In the case no stages have been found, we create the default stages for the user
+        if not stage:
+            self.env['project.task.type'].sudo().with_context(lang=user.partner_id.lang, default_project_id=False).create(
+                self.with_context(lang=user.partner_id.lang)._get_default_personal_stage_create_vals(user.id)
+            )
+        action = self.env['ir.actions.act_window']._for_xml_id('project.action_view_all_task')
+        return action
```

Comment:
```suggestion
        user = self.env.user
        ProjectTaskTypeSudo = self.env['project.task.type'].sudo()
        # In the case no stages have been found, we create the default stages for the user
        if not ProjectTaskTypeSudo.search([('user_id', '=', user.id)], limit=1):
            ProjectTaskTypeSudo.with_context(lang=user.partner_id.lang, default_project_id=False).create(
                self.with_context(lang=user.partner_id.lang)._get_default_personal_stage_create_vals(user.id)
            )
        action = self.env['ir.actions.act_window']._for_xml_id('project.action_view_all_task')
        return action
```
- No need for `stage` variable.
- ProjectTaskTypeSudo makes things a little lighter imho. Up to you!

Reply:
My bad, I sould have been more careful the first time : search_count is enough.